src/ProgressOnderwijsUtils/Data/ParameterizedSqlObjectMapper.cs

```diff
@@ -637,6 +638,46 @@ public static void VerifyDataReaderShape([NotNull] TReader reader)
                 }
             }
         }
+
+        static class BackingFieldDetector
+        {
+            const string BackingFieldPrefix = "<";
+            const string BackingFieldSuffix = ">k__BackingField";
+            const BindingFlags privateInstance = BindingFlags.Instance | BindingFlags.NonPublic;
+            const BindingFlags anyInstance = privateInstance | BindingFlags.Public;
+
+            [NotNull]
+            static string BackingFieldFromAutoPropName([NotNull] string propertyName)
+                => BackingFieldPrefix + propertyName + BackingFieldSuffix;
+
+            [CanBeNull]
+            static string AutoPropNameFromBackingField([NotNull] string fieldName)
+                => fieldName.StartsWith(BackingFieldPrefix, StringComparison.Ordinal) && fieldName.EndsWith(BackingFieldSuffix, StringComparison.Ordinal)
+                    ? fieldName.Substring(BackingFieldPrefix.Length, fieldName.Length - BackingFieldPrefix.Length - BackingFieldSuffix.Length)
+                    : null;
+
+            static bool IsCompilerGenerated([CanBeNull] MemberInfo member)
+                => member != null && member.IsDefined(typeof(CompilerGeneratedAttribute), true);
+
+            static bool IsAutoProp(PropertyInfo autoProperty)
+                => IsCompilerGenerated(autoProperty.GetGetMethod(true));
+
+            [CanBeNull]
+            public static FieldInfo BackingFieldOfPropertyOrNull([NotNull] PropertyInfo propertyInfo)

```

Comment:
Klinkt als iets wat al wel 100 keer gedaan is.

Reply:
Niet in onze codebase iig, en gezien hoe simpel de code is heb ik niet perse zin in een extra dependency hiervoor.